components/shared/BaseButton.tsx

```diff
@@ -0,0 +1,23 @@
+import { FC } from "react";
+
+interface Props {
+    callback: Function;
+    type: 'button' | 'submit';
```

Comment:
you don't need to define this yourself, the button html props already have this defined and you can take advantage of that by extending from them ```interface BaseButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {``` and then passing all props directly to the button tag

Reply:
this saves you from even having to declare type or callback